src/librustc/middle/region.rs

```diff
@@ -110,11 +110,11 @@ pub struct Scope {
     pub(crate) code: u32
 }
 
-const SCOPE_DATA_NODE: u32 = !0;
-const SCOPE_DATA_CALLSITE: u32 = !1;
-const SCOPE_DATA_ARGUMENTS: u32 = !2;
-const SCOPE_DATA_DESTRUCTION: u32 = !3;
-const SCOPE_DATA_REMAINDER_MAX: u32 = !4;
+const SCOPE_DATA_NODE: u32 = 0xFFFF_FFFF;
+const SCOPE_DATA_CALLSITE: u32 = 0xFFFF_FFFE;
+const SCOPE_DATA_ARGUMENTS: u32 = 0xFFFF_FFFD;
+const SCOPE_DATA_DESTRUCTION: u32 = 0xFFFF_FFFC;
+// be sure to add the MAX of FirstStatementIndex if you add more constants here
```

Comment:
Why can't this be done by having an enum with 4 unit variants and a tuple variant around `FirstStatementIndex`?